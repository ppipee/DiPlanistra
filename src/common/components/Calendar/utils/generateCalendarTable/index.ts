import chunk from 'lodash/chunk'

import { CalendarDisplayStateProps } from 'common/components/Calendar/types'
import { NUMBER_OF_DAY_IN_WEEK } from 'common/utils/datetime/constants'

import getFilledDaysInMonth from '../getFilledDaysInMonth'

export default function generateCalendarTable(displayState: CalendarDisplayStateProps) {
	const days = getFilledDaysInMonth(displayState)

	return chunk(days, NUMBER_OF_DAY_IN_WEEK)
}
