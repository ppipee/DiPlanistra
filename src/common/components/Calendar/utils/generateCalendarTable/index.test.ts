import generateCalendarTable from '.'

describe('generateCalendarTable', () => {
	it('should generate calender correctly', () => {
		const calendar = generateCalendarTable({ year: 2020, month: 9 })

		expect(calendar).toMatchSnapshot()
	})
})
