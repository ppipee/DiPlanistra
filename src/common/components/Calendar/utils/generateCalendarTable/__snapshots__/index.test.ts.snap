// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateCalendarTable should generate calender correctly 1`] = `
Array [
  Array [
    2020-09-26T17:00:00.000Z,
    2020-09-27T17:00:00.000Z,
    2020-09-28T17:00:00.000Z,
    2020-09-29T17:00:00.000Z,
    2020-09-30T17:00:00.000Z,
    2020-10-01T17:00:00.000Z,
    2020-10-02T17:00:00.000Z,
  ],
  Array [
    2020-10-03T17:00:00.000Z,
    2020-10-04T17:00:00.000Z,
    2020-10-05T17:00:00.000Z,
    2020-10-06T17:00:00.000Z,
    2020-10-07T17:00:00.000Z,
    2020-10-08T17:00:00.000Z,
    2020-10-09T17:00:00.000Z,
  ],
  Array [
    2020-10-10T17:00:00.000Z,
    2020-10-11T17:00:00.000Z,
    2020-10-12T17:00:00.000Z,
    2020-10-13T17:00:00.000Z,
    2020-10-14T17:00:00.000Z,
    2020-10-15T17:00:00.000Z,
    2020-10-16T17:00:00.000Z,
  ],
  Array [
    2020-10-17T17:00:00.000Z,
    2020-10-18T17:00:00.000Z,
    2020-10-19T17:00:00.000Z,
    2020-10-20T17:00:00.000Z,
    2020-10-21T17:00:00.000Z,
    2020-10-22T17:00:00.000Z,
    2020-10-23T17:00:00.000Z,
  ],
  Array [
    2020-10-24T17:00:00.000Z,
    2020-10-25T17:00:00.000Z,
    2020-10-26T17:00:00.000Z,
    2020-10-27T17:00:00.000Z,
    2020-10-28T17:00:00.000Z,
    2020-10-29T17:00:00.000Z,
    2020-10-30T17:00:00.000Z,
  ],
]
`;
